frontend/src/components/green_house_gaz/materials/DomesticSolidWaste.ts

```diff
@@ -0,0 +1,183 @@
+export function headers(pp_per_hh: number | undefined) {
+  return [
+    ...surveyTableHeaderIncrements,
+    {
+      value: "input.biowaste",
+      text: "Type of waste",
+      items: ["Biowaste", "Non-biowaste"],
+      tooltipInfo:
+        "Definition of biowaste: comprises only biodegradable garden and park waste, food and kitchen waste from households and markets  Source: EAWAG: https://www.eawag.ch/fileadmin/Domain1/Abteilungen/sandec/schwerpunkte/swm/SOWATT/sowatt.pdf",
+      style: {
+        cols: "12",
+      },
+      default: true,
+      type: "select",
+    },
+    {
+      conditional_value: ["Non-biowaste"],
+      conditional: "biowaste",
+      value: "input.biowasteSubCategories",
+      text: "Biowaste subcategories",
+      items: [
+        "Plastics",
+        "Textiles",
+        "Paper / cardboard",
+        "Rubber / leather",
+        "Nappies",
+        "Mixed / unkown composition ",
+      ],
+      style: {
+        cols: "12",
+      },
+      default: true,
+      isInput: true,
+      type: "select",
+      hideFooterContent: false,
+    },
+    {
+      text: "Solid Waste Management (SWM) Practices",
+      value: "input.practiceType",
+      items: ["Open pits, unmanaged", "Managed disposal site", "Open burning"],
+      formatter: (x: string) => x,
+      tooltipInfo: function (value: string) {
+        if (value === "Open pits, unmanaged") {
+          return "Assumed to be shallow < 5 m depth";
+        }
+        if (value === "Managed disposal site") {
+          return "At least one of these conditions: regular cover material (e.g, soil), mechanical compacting, or leveling of the waste. Example: landfill";
+        }
+      },
+      style: {
+        cols: "12",
+      },
+      default: true,
+      isInput: true,
+      type: "select",
+      hideFooterContent: false,
+    },
+    {
+      text: "Percentage of waste category",
+      computeResults: true,
+      value: "input.percentageOfTotalCategories",
+      // conditional_value: "",
+      // conditional: ["fuelType"],
+      style: {
+        cols: "12",
+      },
+      max: 100,
+      subtype: "percent",
+      type: "number",
+      hideFooterContent: false,
+      formatter: (v: number) => {
+        return formatNumber(v, {
+          style: "percent",
+          maximumFractionDigits: 0,
+        });
+      },
+    },
+    ...surveyTableHeaderCO2,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ].map(ensureSurveyTableHeaders);
+}
+
+export function generateComputeItem(
+  projectTotalHH: number
+): (
+  localItemInput: MaterialSolidWasteItemInput,
+  ghgMapRef: ItemReferencesMap
+) => MaterialSolidWasteItemResults {
+  return function computeItem(
+    localItemInput: MaterialSolidWasteItemInput,
+    ghgMapRef: ItemReferencesMap
+  ): MaterialSolidWasteItemResults {
+    const { percentageOfTotalCategories } = localItemInput;
+    if (percentageOfTotalCategories === undefined) {
+      console.log(ghgMapRef);
```

Comment:
console call to remove.

Reply:
ça fait planter typescript si je l'enlève ahah : le temps que je finisse..